File: src/main/java/com/freya02/docs/data/ClassDetailType.java
package com.freya02.docs.data;

public enum ClassDetailType {
	FIELD("field-detail"),
	CONSTRUCTOR("constructor-detail"),
	METHOD("method-detail"),
	ANNOTATION_ELEMENT("annotation-interface-element-detail");

	private final String detailId;
	ClassDetailType(String detailId) {
		this.detailId = detailId;
	}

	public String getDetailId() {
		return detailId;
	}
}

File: doxxy-docs/src/main/kotlin/dev/freya02/doxxy/docs/data/ClassDetailType.kt
package dev.freya02.doxxy.docs.data

enum class ClassDetailType(val detailId: String) {
    FIELD("field-detail"),
    CONSTRUCTOR("constructor-detail"),
    METHOD("method-detail"),
    ANNOTATION_ELEMENT("annotation-interface-element-detail"),
    ENUM_CONSTANTS("enum-constant-detail");
}